backend/write_to_catalog/task.py

```diff
@@ -0,0 +1,57 @@
+import boto3
+from pystac import STAC_IO, Catalog, CatalogType  # type: ignore[import]
+
+from ..api_responses import BODY_KEY
+from ..pystac_io_methods import read_method, write_method
+from ..resources import ResourceName
+from ..types import JsonObject
+
+STAC_IO.write_text_method = write_method
+STAC_IO.read_text_method = read_method
+
+S3_CLIENT = boto3.client("s3")
+
+ROOT_CATALOG_ID = "root_catalog_id"
+ROOT_CATALOG_TITLE = "Geospatial Datalake Root Catalog"
+ROOT_CATALOG_DESCRIPTION = "The root catalog which links to all datasets catalogs in the data lake"
+CATALOG_KEY = "catalog.json"
+
+
+def lambda_handler(_event: JsonObject, _context: bytes) -> JsonObject:
+    """Main Lambda entry point."""
+
+    results = S3_CLIENT.list_objects(
+        Bucket=ResourceName.STORAGE_BUCKET_NAME.value, Prefix=CATALOG_KEY
+    )
+
+    # create root catalog if it doesn't exist
+    if "Contents" not in results:
+        root_catalog = Catalog(
+            id=ROOT_CATALOG_ID,
+            title=ROOT_CATALOG_TITLE,
+            description=ROOT_CATALOG_DESCRIPTION,
+            catalog_type=CatalogType.SELF_CONTAINED,
+        )
+
+    else:
+        root_catalog = Catalog.from_file(
+            f"s3://{ResourceName.STORAGE_BUCKET_NAME.value}/{CATALOG_KEY}"
```

Comment:
There's an `S3_URL_PREFIX` constant you can use for "s3://", although it will need to be pulled out into shared code.

Also, would it be possible to use the `result` above to get the file contents? I expect there should be something like a `results["Contents"][0]["Body"]` which is a file stream.

Reply:
So far as I can see, the list_objects response does not contain file streams.